base/task.jl

```diff
@@ -197,6 +215,7 @@ function produce(v)
     t.state == :runnable || throw(AssertionError("producer.consumer.state == :runnable"))
     if empty
         schedule_and_wait(t, v)
+        ct = current_task() # When a task is copied, ct should be updated to new task ID.
```

Comment:
That is what I mean by running. More precisely a task that have already started but haven't finished. The fact that you need to change this at all means it is silently breaking all kinds of assumptions.